core/src/main/scala/pureconfig/CoproductHint.scala

```diff
@@ -92,6 +92,37 @@ class FieldCoproductHint[T](key: String) extends CoproductHint[T] {
 }
 
 /**
+ * Hint applicable to sealed families of case objects where objects are written and read as strings. Trying to read or
```

Comment:
I would add that the `string` used is the object name, e.g. `object Foo` will be read as `foo` with this. It's not super clear how objects are written and read as strings.